explore/src/main/scala/explore/observationtree/ObsSummaryTable.scala

```diff
@@ -199,10 +202,28 @@ object ObsSummaryTable:
             .setSize(35.toPx),
           mixedColumn(
             TargetColumnId,
-            r => <.span(r.obs.title),
-            r => targetUrl(r.obsId, r.targetWithId)
+            r => r.obs.title,
+            r => (r.obsId, r.targetWithId)
           )
-            .setCell(_.value),
+            .setCell { c =>
+              c.value match {
+                case s: String => <.span(s)
+                case (a, b)    => targetUrl(a, b)
+              }
+            }
+            .setSortingFn { (a, b, c) =>
+              val aʹ: String | (Observation.Id, TargetWithId) = a.getValue(c)
+              val aʹʹ                                         = aʹ match {
+                case s: String => s
+                case (_, b)    => b.target.name.value
+              }
```

Comment:
Maybe this can be in an `extension (name: String | (Observation.Id, TargetWithId))`?

Then we can simplify this to `a.getValue(c).compareTo(b.getValue(c))` (and avoid duplicate logic).